src/lib/Factory/Export/ParametersFactory.php

```diff
@@ -0,0 +1,337 @@
+            throw new InvalidArgumentException(sprintf(
+                'SiteAccess %s doesn\'t exists',
+                $options['siteaccess']
+            ));
+        }
+
+        $configuration = $this->countConfiguredCustomerSettings() === 1
+            ? $this->getCredentialsAndSiteAccessForSingleConfiguration($options)
+            : $this->getCredentialsAndSiteAccessForMultiCustomerConfiguration($options);
+
+        $siteAccess = $configuration['siteaccess'];
+        $customerId = $configuration['customer_id'];
+
+        return [
+            'customer_id' => $customerId,
+            'license_key' => $configuration['license_key'],
+            'siteaccess' => $configuration['siteaccess'],
+            'item_type_identifier_list' => $options['item_type_identifier_list'],
+            'languages' => $options['languages'],
+            'host' => $options['host'] ?? $this->getHostUri($siteAccess),
+            'web_hook' => $options['web_hook'] ?? $this->getWebHook(
+                (int)$customerId,
+                $siteAccess
+            ),
+            'page_size' => $options['page_size'],
+        ];
+    }
+
+    private function hasConfiguredSiteAccess(string $siteAccessName): bool
+    {
+        foreach ($this->siteAccessService->getAll() as $siteAccess) {
+            if ($siteAccessName === $siteAccess->name) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function countConfiguredCustomerSettings(): int
+    {
+        $configuredCounter = 0;
+
+        foreach ($this->siteAccessService->getAll() as $siteAccess) {
+            if ($this->credentialsResolver->hasCredentials($siteAccess->name)) {
+                ++$configuredCounter;
+            }
+        }
+
+        return $configuredCounter;
+    }
+
+    /**
+     * @phpstan-param array{
+     *  item_type_identifier_list: string,
+     *  languages: string,
+     *  page_size: string,
+     *  customer_id: ?string,
+     *  license_key: ?string,
+     *  siteaccess: ?string,
+     *  web_hook: ?string,
+     *  host: ?string,
+     * } $options
+     *
+     * @phpstan-return array{
+     *  customer_id: string,
+     *  license_key: string,
+     *  siteaccess: string,
+     * }
+     *
+     * @throws \EzSystems\EzRecommendationClient\Exception\MissingExportParameterException
+     */
+    private function getCredentialsAndSiteAccessForSingleConfiguration(array $options): array
+    {
+        if (
+            (isset($options['customer_id']) || isset($options['license_key']))
+            && $this->hasMissingRequiredOptions($options)
+        ) {
+            throw new MissingExportParameterException(
+                $this->getMissingRequiredOptions($options),
+                $this->exportParametersType
+            );
+        }
+
+        if (null !== $options['customer_id'] && null !== $options['license_key'] && null !== $options['siteaccess']) {
```

Comment:
Personally I'd use...
```suggestion
        if (isset($options['customer_id'], $options['license_key'], $options['siteaccess']) {
```
...purely because it does not involve negation three times. @alongosz :+1: / :-1: ?

Reply:
> Personally I'd use...
> 
> ...purely because it does not involve negation three times.

:+1: In that specific case I'd agree because of readability (typically we try rather to make the most strict possible comparison).